worker/provisioner/provisioner.go

```diff
@@ -292,6 +302,7 @@ func NewContainerProvisioner(
 	p.Provisioner = p
 	logger.Tracef("Starting %s provisioner for %q", p.containerType, p.agentConfig.Tag())
 
+	p.configObserver.dying = p.catacomb.Dying
```

Comment:
given that SetObserver is only in export_test, I was wondering if it made sense that calling SetObserver would actually peek at the type and grab the raw p.catacomb instead of doing it in New* always, even though it is never used?

Thoughts?